### PR TITLE
Introduce a `position` config for auth filter engagement

### PR DESCRIPTION
**Pull request by ldclakmal**
## Purpose
This PR introduces a `position` configuration, which refers to the index of the authn and authz filters in the `http:Filter` array. The value starts at 1 and can go up to the length of the `http:Filter` array.

Fixes: #17083 

NOTE: Please merge after https://github.com/ballerina-platform/ballerina-lang/pull/17096

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by riyafa** [comment]
Do we need a test?

**Comment by ldclakmal** [comment]
@riyafa since there is no way to add a unit test or integration test for this, I will add a scenario test later. For now, the behaviour is locally verified.

File: stdlib/http/src/main/ballerina/src/http/service_endpoint.bal [hunk 1]
```diff
@@ -177,11 +177,13 @@ public type ServiceEndpointConfiguration record {|
 # should successfully be authorozed.
 # + positiveAuthzCache - The caching configurations for positive authorizations.
 # + negativeAuthzCache - The caching configurations for negative authorizations.
+# + position - The authn/authz filter position of the filter array. The position values starts from 1 and it is set to 1 implicitly.
```
Comment (wggihan) [inline stdlib/http/src/main/ballerina/src/http/service_endpoint.bal]: IMO, this statement gives misleading idea. Ballerina array starts with 0. So in here we say,  we are having filter array, but position start with 1. Can't we align with that normal array positioning?

Comment (ldclakmal) [reply stdlib/http/src/main/ballerina/src/http/service_endpoint.bal]: Yes. Agree on this. It is fixed with https://github.com/ballerina-platform/ballerina-lang/pull/17124/commits/31c544eea1feb656ae2af1a21d0c4e4103057aa3

**Pull request merged**